### PR TITLE
Lowering for _pdist_forward

### PR DESCRIPTION
**Pull request by zpcore**
Resolves https://github.com/pytorch/xla/issues/5885.

### CODE REVIEW

**Comment by zpcore** [comment]
Oh, there is test failure in pytorch size for corner case shape [0,2], [1,2] and [3,0]. Let me make the fix.

File: torch_xla/csrc/ops/ops.cpp [hunk 1]
```diff
@@ -566,6 +567,32 @@ torch::lazy::NodePtr Norm(const torch::lazy::Value& input,
...
+                      LoweringContext* loctx) -> XlaOpVector {
+    xla::XlaOp xla_input = loctx->GetOutputOp(node.operand(0));
+    return node.ReturnOp(BuildUpperTriangle(xla_input), loctx);
+  };
+  auto lower_for_shape_fn =
+      [&](absl::Span<const xla::XlaOp> operands) -> xla::XlaOp {
+    return BuildUpperTriangle(operands[0]);
+  };
+  torch::lazy::NodePtr tmp = input - torch::lazy::MakeNode<Unsqueeze>(input, 1);
+  torch::lazy::NodePtr result_matrix = Norm(tmp, p, dtype, {2}, false);
```
Comment (wonjoolee95) [inline torch_xla/csrc/ops/ops.cpp:585]: The logic LGTM, but typically we organize this logic in such a way that we have a function called `BuildPdistForward` (instead of `BuildUpperTriangle`) to have all the lowering-specific logic in that function instead of here. Can we do that for this as well?

Comment (wonjoolee95) [reply torch_xla/csrc/ops/ops.cpp:585]: Understood, makes sense. I was referring to putting the lines 584-585 in that `BuildPdistForward` as well.

Comment (zpcore) [reply torch_xla/csrc/ops/ops.cpp:585]: I see, this case is a bit tricky. Line 584-585 were handled in node(`torch::lazy::NodePtr`) level using the existing `Unsqueeze` and `Norm` Nodes, while the remaining is handled in lower xla op level since they don't provide `Norm` etc. as we needed here. I didn't see others handle `torch::lazy::NodePtr` in `Build*` functions. That's why I split them out.

Comment (wonjoolee95) [reply torch_xla/csrc/ops/ops.cpp:585]: I see, sg, let's just leave it as is. Thanks!

**Review by wonjoolee95 — APPROVED** [review]
LGTM, thanks!
